pkg/e2e/infra/infra.go

```diff
@@ -197,231 +246,73 @@ var _ = g.Describe("[Feature:Machines] Managed cluster should", func() {
-		randomUUID := string(uuid.NewUUID())
-		for i := 0; i < scaleOut; i++ {
-			targetMachineSet := existingMachineSets[i%len(existingMachineSets)]
-			machineSetName := fmt.Sprintf("e2e-%s-w-%d", randomUUID[:5], i)
-			glog.Infof("Creating transient MachineSet %q", machineSetName)
-			machineSets[i] = e2e.NewMachineSet(
-				targetMachineSet.Labels[e2e.ClusterKey],
-				targetMachineSet.Namespace,
-				machineSetName,
-				targetMachineSet.Spec.Selector.MatchLabels,
-				targetMachineSet.Spec.Template.ObjectMeta.Labels,
-				&targetMachineSet.Spec.Template.Spec.ProviderSpec,
-				0) // zero replicas
-			machineSets[i].Spec.Template.Spec.ObjectMeta.Labels = map[string]string{
-				e2e.WorkerNodeRoleLabel: "",
-			}
-			o.Expect(client.Create(context.TODO(), machineSets[i])).Should(o.Succeed())
-			cleanupObjects = append(cleanupObjects, runtime.Object(machineSets[i]))
-		}
-
-		g.By(fmt.Sprintf("scaling %q from %d to %d replicas", machineSets[0].Name, 0, scaleOut))
-		err = scaleMachineSet(machineSets[0].Name, scaleOut)
-		o.Expect(err).NotTo(o.HaveOccurred())
-
-		g.By(fmt.Sprintf("scaling %q from %d to %d replicas", machineSets[1].Name, 0, scaleOut))
-		err = scaleMachineSet(machineSets[1].Name, scaleOut)
-		o.Expect(err).NotTo(o.HaveOccurred())
-
-		o.Eventually(func() bool {
-			nodes0, err := GetNodesFromMachineSet(client, machineSets[0])
-			if err != nil {
-				return false
-			}
-			nodes1, err := GetNodesFromMachineSet(client, machineSets[1])
-			if err != nil {
-				return false
-			}
-			return len(nodes0) == scaleOut && NodesAreReady(nodes0) && len(nodes1) == scaleOut && NodesAreReady(nodes1)
-		}, e2e.WaitLong, 5*time.Second).Should(o.BeTrue())
-
-		g.By(fmt.Sprintf("scaling %q from %d to %d replicas", machineSets[0].Name, scaleOut, 0))
-		err = scaleMachineSet(machineSets[0].Name, 0)
-		o.Expect(err).NotTo(o.HaveOccurred())
-
-		g.By(fmt.Sprintf("scaling %q from %d to %d replicas", machineSets[1].Name, scaleOut, 0))
-		err = scaleMachineSet(machineSets[1].Name, 0)
-		o.Expect(err).NotTo(o.HaveOccurred())
-
-		g.By(fmt.Sprintf("waiting for cluster to get back to original size. Final size should be %d nodes", existingClusterSize))
-		err = waitForClusterSizeToBeHealthy(client, existingClusterSize)
-		o.Expect(err).NotTo(o.HaveOccurred())
+		e2e.WaitForMachineSet(client, machineSet2.GetName())
+		scaleMachineSet(machineSet.GetName(), 0)
+		scaleMachineSet(machineSet2.GetName(), 3)
+		e2e.WaitForMachineSet(client, machineSet.GetName())
+		e2e.WaitForMachineSet(client, machineSet2.GetName())
 	})
 
 	g.It("drain node before removing machine resource", func() {
-		var err error
-		client, err := e2e.LoadClient()
-		o.Expect(err).NotTo(o.HaveOccurred())
+		g.By("Create a machine for node about to be drained")
 
-		g.By("checking existing cluster size")
-		existingClusterSize, err := getClusterSize(client)
-		o.Expect(err).NotTo(o.HaveOccurred())
+		selector := machineSet.Spec.Selector
+		machines, err := e2e.GetMachines(client, &selector)
```

Comment:
This test expects machines to contain at least two items, should check that `o.Expect(len(machines)).To(o.BeNumerically(">=", 2))`